src/main/java/codeu/controller/BotServlet.java

```diff
@@ -0,0 +1,149 @@
+package codeu.controller;
+
+
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.data.Conversation;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Servlet class responsible for the BotChat page. */
+public class BotServlet extends HttpServlet {
+
+
+    public static final String DEFULT_BOT_CONVERSATION_TITLE="Bot-Conversation ";
+
+    /**
+     * Store class that gives access to Messages.
+     */
+    private MessageStore messageStore;
+
+    /**
+     * Store class that gives access to Users.
+     */
+    private UserStore userStore;
+
+    /** Store class that gives access to Conversations. */
+    private ConversationStore conversationStore;
+
+    /**
+     * Set up state for handling chat requests.
+     */
+
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setMessageStore(MessageStore.getInstance());
+        userStore = UserStore.getInstance();
+        conversationStore = ConversationStore.getInstance();
+    }
+
+    /**
+     * Sets the MessageStore used by this servlet. This function provides a common setup method for
+     * use by the test framework or the servlet's init() function.
+     */
+    void setMessageStore(MessageStore messageStore) {
+        this.messageStore = messageStore;
+    }
+    /**
+     * This function fires when a user navigates to the bot chat page. It gets the conversation title "bot-conversation" from
+     * the URL, finds the corresponding Conversation, and fetches the messages in that Conversation.
+     * It then forwards to bot chat.jsp for rendering.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+        String username = (String) request.getSession().getAttribute("user");
+        if (username == null) {
+            //redirect to login
+            response.sendRedirect("/login");
+            return;
+        }
+        User user = userStore.getUser(username);
+        if (user == null) {
+            // couldn't find author, redirect to loging Page
+            response.sendRedirect("/login");
+            return;
+        }
+
+        Conversation botConversation = conversationStore.getBotConversation(user.getId());
+        if (botConversation == null){
+             botConversation = new Conversation(UUID.randomUUID(), user.getId(), DEFULT_BOT_CONVERSATION_TITLE+username, Instant.now());
```

Comment:
I know it's highly unlikely, but it's possible that the randomly generated UUID will be the same as a previously stored object's, so it might be a good idea to double check this before creating the new conversation.

Reply:
will check it, if i could come up with any other idea.